File: algorithms/Strings/Medium/MultiplyString/VSDMultiplyString.py

def multiply(num1,num2):
    res=0
    ans=""

    for i in range(len(num1)-1,-1,-1):
        x=int(num1[i])
        ans=""
        carry=0
        for j in range(len(num2)-1,-1,-1):
            y=int(num2[j])
            product=((x*y)+carry)
            carry=product//10
            ans+=str(product%10)
        x="0"*(len(num1)-i-1)
        ans=x+ans 
        if carry:
            ans+=str(carry)
        res+=int(ans[::-1])
    return str(res)

if __name__=="__main__":
    num1="123"
    num2="456"
    print(multiply(num1,num2))

File: algorithms/Strings/Medium/MultiplyString/VSDMultiplyString.cpp

#include <bits/stdc++.h>

std::string multiply(std::string num1, std::string num2)
{
    int m = num1.length(), n = num2.length();
    int pos[m + n]{0};

    for (int i = m - 1;i>=0;i--){
        for(int j=n-1;j>=0;j--){
            int mul = (num1[i] - '0') * (num2[j]-'0');
            int p1 = i+j, p2 = i+j+1;
            int sum = mul + pos[p2];
            
            pos[p1] += sum/10;
            pos[p2] = (sum) % 10;
        }
    }
    std::string ans = "";
    for(int p :pos) 
        if(!(ans.length() == 0 && p==0))ans+=std::to_string(p);
    
    return ans.length() == 0 ? "0" : ans;
}

int main()
{
    std::string num1 = "123";
    std::string num2 = "456";
    std::cout << multiply(num1, num2);
    return 0;
}